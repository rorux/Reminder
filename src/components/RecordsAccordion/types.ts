export type TRecordAccordion = {
  period: string;
  title: string;
  id: string;
  comment: string;
};

export type TRecordsAccordionProps = {
  recordListRender: Array<TRecordAccordion>;
};
