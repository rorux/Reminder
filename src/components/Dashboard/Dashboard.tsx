import React from 'react';

const Dashboard = () => {
  return <div data-test="dashboard">Дашборд</div>;
};

export default Dashboard;
