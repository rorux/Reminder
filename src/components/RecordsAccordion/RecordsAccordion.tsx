import React from 'react';
import { TRecordsAccordionProps } from './types';
import RecordItem from '@components/RecordItem';

const RecordsAccordion: React.FC<TRecordsAccordionProps> = ({ recordListRender, handleDelete }) => {
  const [expanded, setExpanded] = React.useState<string | false>(false);

  const handleChangeToggle =
    (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
      setExpanded(isExpanded ? panel : false);
    };

  return (
    <>
      {recordListRender.map((record) => (
        <RecordItem
          record={record}
          key={record.id}
          handleChange={handleChangeToggle}
          handleDelete={handleDelete}
          expanded={expanded}
        />
      ))}
    </>
  );
};

export default RecordsAccordion;
