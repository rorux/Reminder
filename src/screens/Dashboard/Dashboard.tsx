import React from 'react';

const Dashboard = () => {
  return (
    <div className="dashboard" data-test="dashboardComponent">
      Дашборд
    </div>
  );
};

export default Dashboard;
