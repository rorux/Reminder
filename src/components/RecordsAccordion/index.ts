import RecordsAccordion from './RecordsAccordion';

export default RecordsAccordion;
