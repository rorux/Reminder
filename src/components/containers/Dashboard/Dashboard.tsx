import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Typography from '@mui/material/Typography';
import Fab from '@mui/material/Fab';
import AddIcon from '@mui/icons-material/Add';
import Tooltip from '@mui/material/Tooltip';
import Grid from '@mui/material/Grid';
import { ROUTES } from '@router/types';
import RecordsAccordion from '@components/RecordsAccordion';
import { initRecords } from '@store/records/actions';
import { recordsSelector } from '@store/records/selectors';
import RecordListRenderer from '@utils/records/RecordListRenderer';

const Dashboard = () => {
  const { recordList } = useSelector(recordsSelector);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(initRecords());
  }, [dispatch]);

  const recordListRender = RecordListRenderer(recordList);

  return (
    <div data-test="dashboard">
      <Grid container>
        <Grid item sm>
          <Typography variant="overline" display="block" gutterBottom>
            Список дел
          </Typography>
        </Grid>
        <Grid item>
          <Link to={ROUTES.CREATE_RECORD}>
            <Tooltip title="Создать новое напоминание" placement="left">
              <Fab
                size="small"
                color="inherit"
                aria-label="add"
                sx={{ mb: 1, boxShadow: 0 }}
                variant="extended"
              >
                <AddIcon />
              </Fab>
            </Tooltip>
          </Link>
        </Grid>
      </Grid>

      {recordListRender.length ? (
        <RecordsAccordion recordListRender={recordListRender} />
      ) : (
        <Typography variant="body2" gutterBottom>
          У Вас нет сохраненных записей. Создайте напоминание, кликнув на &quot;плюс&ldquo; в правом
          верхнем углу.
        </Typography>
      )}
    </div>
  );
};

export default Dashboard;
