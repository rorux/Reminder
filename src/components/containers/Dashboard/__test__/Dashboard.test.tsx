import * as React from 'react';
import * as Redux from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import Dashboard from '../Dashboard';
import RecordsAccordion from '@components/RecordsAccordion';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(() => ({
    recordList: [
      {
        id: '1',
        title: 'title of monthly remind',
        period: 'monthly',
        monthday: '8',
        holidays: 'afterHoliday',
        days: [1642366800000, 1644872400000],
      },
      {
        id: '2',
        title: 'title of quarterly remind',
        period: 'quarterly',
        monthday: '8',
        quarter: 'Янв, Апр, Июл, Окт',
        holidays: 'beforeHoliday',
        days: [1642366800000, 1644872400000],
      },
    ],
  })),
  // useDispatch: jest.fn(),
}));

const setUp = () => shallow(<Dashboard />);

describe('Dashboard component', () => {
  let component: Enzyme.ShallowWrapper;
  let useDispatchSpy: any;
  let mockDispatchFn: any;
  beforeEach(() => {
    useDispatchSpy = jest.spyOn(Redux, 'useDispatch');
    mockDispatchFn = jest.fn();
    useDispatchSpy.mockReturnValue(mockDispatchFn);
    component = setUp();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Snapshot of Dashboard component', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render RecordListRenderer component', () => {
    expect(component.find(RecordsAccordion)).toHaveLength(1);
    expect(useDispatchSpy).toHaveBeenCalled();
  });
});
