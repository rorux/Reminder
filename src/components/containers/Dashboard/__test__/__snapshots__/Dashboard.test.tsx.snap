// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Snapshot of Dashboard component 1`] = `
<Fragment>
  <div
    data-test="dashboard"
  >
    <ForwardRef(Grid)
      container={true}
    >
      <ForwardRef(Grid)
        item={true}
        sm={true}
      >
        <ForwardRef(Typography)
          display="block"
          gutterBottom={true}
          variant="overline"
        >
          Список дел
        </ForwardRef(Typography)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <Link
          to="/create"
        >
          <ForwardRef(Tooltip)
            placement="left"
            title="Создать новое напоминание"
          >
            <ForwardRef(Fab)
              aria-label="add"
              color="inherit"
              size="small"
              sx={
                Object {
                  "boxShadow": 0,
                  "mb": 1,
                }
              }
              variant="extended"
            >
              <Memo(ForwardRef(AddIcon)) />
            </ForwardRef(Fab)>
          </ForwardRef(Tooltip)>
        </Link>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
    <RecordsAccordion
      handleDelete={[Function]}
      recordListRender={
        Array [
          Object {
            "comment": "8-го числа. В случае выпадения на выходной или праздничный день перенос на ближайший рабочий день ПОСЛЕ наступления срока.",
            "id": "1",
            "period": "Ежемесячно",
            "title": "title of monthly remind",
          },
          Object {
            "comment": "8-го числа в январе, апреле, июле, октябре. В случае выпадения на выходной или праздничный день перенос на ближайший рабочий день ДО наступления срока.",
            "id": "2",
            "period": "Ежеквартально",
            "title": "title of quarterly remind",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
